chore: Convert browser tab to typescript and fix spoofing

tommasini opened the pull request:
## **Description**

Refactor of browser tab to typescript 

Please visit this issue that includes a bigger description: https://github.com/MetaMask/mobile-planning/issues/2087


## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

tommasini committed:
Convert browser tab to typescript

tommasini committed:
fix snapshot

tommasini committed:
fix snapshot

Cal-L committed:
Abstract types, consts, and styles

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Store original browser tab in BrowserTab file

Cal-L committed:
Create simplified webview that prevents spoofing issues

Cal-L committed:
Clean up code

Cal-L committed:
Also handle spoof scenarios for Android

tommasini committed:
small fix, import react on simple webview

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

tommasini committed:
types and eslint browser tab final version

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
update browser tab utils Url variable name

Cal-L committed:
Replace browser tab input

Cal-L committed:
Handle condition where load ends with same origins

tommasini committed:
move to browser tab  utils getMaskedUrl and isEnsUrl function

joaoloureirop committed:
fix back to safety button on phishing modal

Cal-L committed:
Update origin name to prevent type conflicts

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

joaoloureirop committed:
inApp Browser: fix home button

tommasini committed:
isolated phishing modal

Cal-L committed:
Clean up AccountRightButton press tracking logic

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

tommasini committed:
abstract options logic from browser tab component

tommasini committed:
Abstract and test IpfsBanner in BrowserTab

Cal-L committed:
Move elements into BrowserUrlBar

Cal-L committed:
Remove browser nav bar

Cal-L committed:
Convert UrlAutocomplete to TS

Cal-L committed:
Remove setting browser navigation bar

Cal-L committed:
Remove redundant BrowserTab

Cal-L committed:
Update types and styles of BrowserTab

Cal-L committed:
Consolidate BrowserUrlBar

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

tommasini committed:
go to homepage

tommasini committed:
active url to save with useState

joaoloureirop committed:
fix browser back & forward nav

tommasini committed:
fix eslint add dependency activeUrl to sendAllNotifications function and remove onLoad from handleWebviewNavigationChange switch function

tommasini committed:
remove unneeded onLoad property from webview

tommasini committed:
replace go for onSubmitEditing function at goToHomePage and returnHome functions

tommasini committed:
refactor PhishingModal go property

tommasini committed:
remove console log from options

Cal-L committed:
Make connection icon more intuitive

Cal-L committed:
Reset progress bar when website cannot be reached

Cal-L committed:
Remove duplicate sanitize function for browser url

tommasini committed:
portfolio new tab with metrics and marketing stored value to not repeat the metrics modal

Cal-L committed:
Add blank SVG

Cal-L committed:
Fix fallback display url when website cannot be loaded

Cal-L committed:
Remove unused code in Browser

Cal-L committed:
Update browser tabs to use AvatarFavicon

Cal-L committed:
Make ref names more obvious

Cal-L committed:
Fix favicon fallback when uri is missing from remote source

Cal-L committed:
Defer loading webview until focused

Cal-L committed:
Resolve conflicts

Cal-L committed:
Add missing enum

brianacnguyen commented on app/component-library/components/Icons/Icon/Icon.assets.ts:
```diff
@@ -167,6 +167,7 @@ import walletmoneySVG from './assets/wallet-money.svg';
 import walletSVG from './assets/wallet.svg';
 import warningSVG from './assets/warning.svg';
 import wifiSVG from './assets/wifi.svg';
+import blankSVG from './assets/blank.svg';
```
@tommasini why was this needed?

tommasini replied:
This was added by @Cal-L on this [commit ](https://github.com/MetaMask/metamask-mobile/pull/12740/commits/14796932b38da816b6f3daa39503e1be4b6b85ff)

Cal-L replied:
It was used to hide the connection type icon in the address bar. Not needed anymore, so removed, thanks!

brianacnguyen reviewed (changes requested):
a comment

tommasini committed:
handle ens ipfs content fixed

Cal-L committed:
Hide reload and favorites for undefined url

Cal-L committed:
Fix handling error

Cal-L committed:
Reset error when going back

Cal-L committed:
Remove old browser url modal

Cal-L committed:
Remove unused params from browser nav

Cal-L committed:
Abstract style from Browser

Cal-L committed:
Remove unused event

Cal-L committed:
Properly handle connection icon state

Cal-L committed:
Add clear button to address bar

Cal-L committed:
Hide bottom bar when focusing input

Cal-L committed:
Fix url unexpectedly changing after hitting submit

Cal-L committed:
Fix persisting submitted entered and autocomplete urls when address bar is unfocused

Cal-L committed:
Clean go home logic

Cal-L committed:
Force resolve url when using webview back button

tommasini committed:
remove go function and using onSubmitEditing, also refactores initial url logic to simplify functions responsabilities

tommasini committed:
s

tommasini committed:
handleFirst Url after getEntryScriptWeb3

Cal-L committed:
Remove unused ens code from browserUrlBar

Cal-L committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Fix lint issues

Cal-L committed:
Handle undefined ENS conditions

Cal-L committed:
Fix lint issue

Cal-L committed:
Fix Favicon tests

Cal-L committed:
Update snapshots

Cal-L committed:
Fix browser util test

Cal-L committed:
Write unit tests for browser tab utils

Cal-L committed:
Fix snapshots for mocked png

Cal-L committed:
Fix lint issues in tests

Cal-L committed:
Ignore import/no-commonjs for png mock

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

Cal-L committed:
Pass correct params to portfolio homepage

Cal-L committed:
Fix lint issues

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Simplify bridge creation logic on BrowserTab

Cal-L committed:
Simplify component did mount

tommasini committed:
fix ts issue

Cal-L committed:
Merge branch 'main' into fix/browser-tab-convert-ts

Cal-L committed:
Merge branch 'fix/browser-tab-convert-ts' of github.com:MetaMask/metamask-mobile into fix/browser-tab-convert-ts

Cal-L committed:
Remove blank icon

Cal-L committed:
Remove remaining blank name

tommasini committed:
add snapshot testing to browserUrlBar

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
replacing resolvedUrlRef for a useState resolvedUrl, so it can grab the value when checking tabPermissions, when it was a ref when the value updated the function was using the original value, creating an error

tommasini committed:
add missing dependencies to useCallback that were missing

tommasini committed:
add snapshot test for when the browserUrlBar is not focused

tommasini committed:
add snapshot for DeleteWalletModal component

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
default browser page to google

tommasini committed:
add 4 seconds to wait for tapping network avatar button

tommasini committed:
increase unit test coverage of browser url bar component

tommasini committed:
revert to use the ref of urlResolvedRed instead of state to reduce re renders of the browser and remove delay to try on e2e with new solution to checkTabPermissions

tommasini committed:
address code quality issues detected by sonar cloud; refactor update allow list ref with use effect, and considering to remove it completely and use only the redux state; update ternary in styles to a function to increase readability

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
re add delay of 4 seconds to wait to tap on network avatar while connected icon in e2e

tommasini committed:
chore: Refactor whitelist inapp browser (#13049)

## **Description**

Refactor whitelist inapp browser to use directly the in-app browser.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

EtherWizard33 reviewed (dismissed):
approved for wallet-ux on the two following files codeownered:
app/components/UI/OnboardingWizard/index.tsx
app/components/Views/Settings/SecuritySettings/SecuritySettings.tsx

adonesky1 reviewed (dismissed):
RPCMethodMiddleware changes look good to me!

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

metamaskbot committed:
Bump version number to 1534

Cal-L reviewed (approved):
Lgtm!

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

tommasini committed:
Merge branch 'main' into fix/browser-tab-convert-ts

EtherWizard33 reviewed (approved):
re-approve after prev was reset

tommasini commented on app/actions/browser/index.js:
```diff
@@ -36,10 +36,12 @@ export function addToHistory({ url, name }) {
   return {
     type: 'CLEAR_BROWSER_HISTORY',
     id: Date.now(),
+    metricsEnabled,
```
Yes just for simplicity let me cover it in another PR!

christopherferreira9 reviewed (approved):
re-approve from @abretonc7s 's
